scripts/validate_string_concatenation.py

```diff
@@ -1,51 +1,293 @@
+
+    For example, this is wrong:
+
+    >>> with pytest.raise(ValueError):
+    ...     # Some code that raises ValueError
+
+    And this is what we want instead:
+
+    >>> with pytest.raise(ValueError, match="foo"):
+    ...     # Some code that raises ValueError
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    msg : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #23922
+    """
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for counter, current_token in enumerate(tokens, start=1):
+        if not (current_token.type == token.NAME and current_token.string == "raises"):
+            continue
+        for next_token in tokens[counter:]:
+            if next_token.type == token.NAME and next_token.string == "match":
+                break
+            # token.NEWLINE refers to the end of a logical line
+            # unlike token.NL or "\n" which represents a newline
+            if next_token.type == token.NEWLINE:
+                yield (
+                    source_path,
+                    current_token.start[0],
+                    "Bare pytests raise have been found. "
+                    "Please pass in the argument 'match' as well the exception.",
+                )
+                break
+
+
+def strings_to_concatenate(
+    source_path: str,
+) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    This test case is necessary after 'Black' (https://github.com/psf/black),
+    is formating strings over multiple lines.
+
+    For example, when this:
+
+    >>> foo = (
+    ...     "bar "
+    ...     "baz"
+    ... )
+
+    Is becoming this:
+
+    >>> foo = ("bar " "baz")
+
+    'Black' is not considering this as an
+    issue (see https://github.com/psf/black/issues/1051),
+    so we are checking it here instead.
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    msg : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #30454
+    """
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for current_token, next_token in zip(tokens, tokens[1:]):
+        if current_token.type == next_token.type == token.STRING:
+            yield (
+                source_path,
+                current_token.start[0],
+                (
+                    "String unnecessarily split in two by black. "
+                    "Please merge them manually."
+                ),
+            )
+
+
+def strings_with_wrong_placed_whitespace(
```

Comment:
Are there tests for these functions?

Reply:
No there are no tests for these functions.

I have no clue how to tests these, If you can give me a minimal starting point (how to test one of these functions, or an abstract idea on how to test them) I believe I will be good to go and test every other function and expand the testing even further, I just really don't know where to start.

Reply:
You can add a new file in `scripts/tests/`, check the docstrings one for reference.

Then you can start with a simple test for each function. A function with an assert, something like this for your functions:
```python
def test_sum_works():
    assert sum(1, 2) == 3
```

Then we can review or add comments, or if you feel confident, you can directly test several uses cases, similar to the examples you wrote for the wrong placed whitespaces.

Reply:
@datapythonista ty for the quick reply :)

Do you mind if I will create in addition to the test file, another file (possibly 3) that will contain some data for the tests? because the functions only rake one parameter and that's a file path.

Reply:
That should be ok, but probably better to use `io.StringIO`. Making the functions accept a file descriptor instead of a file path is not a bad idea either. I think it's common, and it can be useful not only for the tests, but for validating content that is not in the file system, but in a url, or somewhere else.

In any case, feel free to implement whatever you think it makes more sense. But I think it's better to write unit tests, testing minimalistic and specific cases, than have files on disk.

Reply:
I don't think should need to create additional files. Should be able to use https://github.com/pandas-dev/pandas/blob/6b87716e1416061b4bf0109a89caf962acdf8c12/scripts/tests/test_validate_docstrings.py#L16 as an example but create another class that has methods exhibiting the behavior you are trying to correct here instead of docstrings